### PR TITLE
std.math: ARM, AArch64, and other __asm compile fixes

### PR DESCRIPTION
**Pull request by joakim-noah**
With the recent commits upstream to make more of `std.math` `@safe` and `pure`, I had to use this patch to get ldc's version to compile on ARM and AArch64. I went ahead and marked up the `__asm` blocks for the other arches in `clearExceptions` too, since they'll need to be `@trusted` to compile also. This pull was tested on Android/ARM, will check linux/AArch64 next.

### CODE REVIEW

File: std/math.d [hunk 1]
```diff
@@ -5901,22 +5901,30 @@ public:
             return true;
         else version(AArch64)
         {
+            bool result;
             auto oldState = getControlState();
-            // If exceptions are not supported, we set the bit but read it back as zero
-            // https://sourceware.org/git/?p=glibc.git;a=blob;f=sysdeps/aarch64/fpu/feenablxcpth.c
-            setControlState(oldState | (divByZeroException & allExceptions));
-            immutable result = (getControlState() & allExceptions) != 0;
-            setControlState(oldState);
```
Comment (joakim-noah) [inline std/math.d]: `setControlState()` cannot be pure in general, so I extracted the `__asm` instructions from there and, [as long as we're only setting exception bits, `hasExceptionTraps()` can still be considered pure (see point 8)](https://dlang.org/spec/function.html#pure-functions).

Comment (ibuclaw) [reply std/math.d]: Actually, as per the concession in the language spec, `setControlState()` _can_ be pure also.  As the FPControl itself saves state in `initialize()` and clears and restores state in `~this()`.

Comment (ibuclaw) [reply std/math.d]: Actually, as per the concession in the language spec, `setControlState()` can be pure also. As the FPControl itself saves state in `initialize()` and clears and restores state in `~this()`.

Comment (joakim-noah) [reply std/math.d]: These are all static functions though, so the struct is not initialized or deconstructed when they're called?

Comment (ibuclaw) [reply std/math.d]: Only `hasExceptionTraps()` is static, all others require a state object.

**Comment by joakim-noah** [comment]
Updated this pull to use Martin's new `__asm_trusted` construct, ldc-developers/druntime#142, and tested it on Android/ARM with ldc 1.10 and linux/AArch64 and Android/AArch64 with ltsmaster, by adding the `pure`/`@safe` attributes to these functions like upstream.

File: std/math.d (hunk 1 above)
Comment (joakim-noah) [reply std/math.d]: Well, `hasExceptionTraps()` is the function in question, so it cannot rely on the struct resetting these flags.

**Pull request merged**